src/qibolab/instruments/erasynth.py

```diff
@@ -11,204 +11,111 @@
-        else:
-            self._post("rfoutput", 1)
-
-    def off(self):
-        if not self.ethernet:
-            self.device.off()
-        else:
-            self._post("rfoutput", 0)
-
-    def close(self):
-        self.is_connected = False
-
-    def _post(self, name, value):
-        """
-        Post a value to the instrument's web server.
-
-        Try to post three times, waiting for 0.1 seconds between each attempt.
+        Try to post multiple times, waiting for 0.1 seconds between each attempt.
 
         Args:
             name: str = The name of the value to post.
             value: str = The value to post.
         """
         value = str(value)
-        for _ in range(MAX_RECONNECTION_ATTEMPTS):
+        for _ in range(self.MAX_RECONNECTION_ATTEMPTS):
             try:
-                response = requests.post(f"http://{self.address}/", data={name: value}, timeout=TIMEOUT)
+                response = requests.post(self.url, data={name: value}, timeout=self.TIMEOUT)
                 if response.status_code == 200:
                     return True
                 break
             except (ConnectionError, TimeoutError, requests.exceptions.ReadTimeout):
                 log.info("ERAsynth connection timed out, retrying...")
-        raise InstrumentException(self, f"Unable to post {name}={value} to {self.name}")
+        raise ConnectionError(f"Unable to post {name}={value} to {self.name}")
 
-    def _get(self, name):
-        """
-        Get a value from the instrument's web server.
+    def get(self, name):
+        """Get a value from the instrument's web server.
 
-        Try to get three times, waiting for 0.1 seconds between each attempt.
+        Try to get multiple times, waiting for 0.1 seconds between each attempt.
 
         Args:
             name: str = The name of the value to get.
         """
-        for _ in range(MAX_RECONNECTION_ATTEMPTS):
-            try:
-                response = requests.post(f"http://{self.address}/", params={"readAll": 1}, timeout=TIMEOUT)
+        if name == "ref_osc_source":
+            value = self.get("reference_int_ext")
+            if value == 1:
+                return "EXT"
+            else:
+                return "INT"
 
+        for _ in range(self.MAX_RECONNECTION_ATTEMPTS):
+            try:
+                response = requests.post(self.url, params={"readAll": 1}, timeout=self.TIMEOUT)
                 if response.status_code == 200:
                     # reponse.text is a dictonary in string format, convert it to a dictonary
                     return json.loads(response.text)[name]
                 break
             except (ConnectionError, TimeoutError, requests.exceptions.ReadTimeout):
                 log.info("ERAsynth connection timed out, retrying...")
-        raise InstrumentException(self, f"Unable to get {name} from {self.name}")
+
+        raise ConnectionError(f"Unable to get {name} from {self.name}")
+
+    def set(self, name, value):
```

Comment:
Missing docstring.
Moreover, the available options (for `name`s, and the corresponding `value`s) should be somehow documented.

Possible alternatives:
- just spell them out in the docstring
- add an `Enum`, with variants documenting the name and accepted values
- [`@overload`](https://docs.python.org/3/library/typing.html?highlight=overload#typing.overload) the `set` function and use the type system (you could use `Literal[...]` for the name, and then the type you wish for the `value`)